java/test/org/openqa/selenium/grid/sessionqueue/local/LocalNewSessionQueueTest.java

```diff
@@ -166,6 +167,97 @@ private void waitUntilAddedToQueue(SessionRequest request) {
                             sessionRequestCapability.getRequestId().equals(r.getRequestId())));
   }
 
+  @ParameterizedTest
+  @MethodSource("data")
+  void testCompleteWithCreatedSession(Supplier<TestData> supplier) {
+    setup(supplier);
+
+    AtomicBoolean isCompleted = new AtomicBoolean(false);
+
+    new Thread(
+            () -> {
+              waitUntilAddedToQueue(sessionRequest);
+
+              Capabilities capabilities = new ImmutableCapabilities("browserName", "chrome");
+              SessionId sessionId = new SessionId("123");
+              Session session =
+                  new Session(
+                      sessionId,
+                      URI.create("https://example.com"),
+                      CAPS,
+                      capabilities,
+                      Instant.now());
+              CreateSessionResponse sessionResponse =
+                  new CreateSessionResponse(
+                      session,
+                      JSON.toJson(
+                              ImmutableMap.of(
+                                  "value",
+                                  ImmutableMap.of(
+                                      "sessionId", sessionId,
+                                      "capabilities", capabilities)))
+                          .getBytes(UTF_8));
+
+              isCompleted.set(queue.complete(sessionRequest.getRequestId(), Either.right(sessionResponse)));
+            })
+        .start();
+
+    HttpResponse httpResponse = queue.addToQueue(sessionRequest);
+    try {
+      Thread.sleep(400); // wait for thread to complete
+    } catch (InterruptedException ignore) {
+    }
```

Comment:
Please use a `CountDownLatch` instead of a `Thread.sleep`. Other tests can serve you as example.

Reply:
ok, thanks